test/e2e/envvars_test.go

```diff
@@ -81,18 +83,78 @@ func TestEnvVars(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Error fetching Route %s: %v", names.Route, err)
 	}
-	domain := route.Status.Domain
 
-	envVarsExpectedOutput := fmt.Sprintf("Here are our env vars service:%s - configuration:%s - revision:%s", names.Service, names.Config, names.Revision)
+	domain := route.Status.Domain + "/envvars/should"
 
-	_, err = pkgTest.WaitForEndpointState(
+	resp, err := pkgTest.WaitForEndpointState(
 		clients.KubeClient,
 		logger,
 		domain,
-		pkgTest.Retrying(pkgTest.MatchesBody(envVarsExpectedOutput), http.StatusNotFound),
+		pkgTest.Retrying(verifyUpStatus, http.StatusNotFound),
 		"EnvVarsServesText",
 		test.ServingFlags.ResolvableDomain)
 	if err != nil {
-		t.Fatalf("The endpoint for Route %s at domain %s didn't serve the expected text \"%s\": %v", names.Route, domain, envVarsExpectedOutput, err)
+		t.Fatalf("Failed before reaching desired state : %v", err)
+	}
+
+	expectedOutputs := map[string]string{
+		"K_SERVICE" : names.Service,
+		"K_CONFIGURATION" : names.Config,
+		"K_REVISION" : names.Revision,
+	}
+	// We are failing even for "SHOULD".
```

Comment:
1. The runtime contract tests is a better fit for conformance, not e2e, directory wise.
2. And it's better have separate tests to validate MUST and SHOULD respectively so that it's super clear that which part doesn't conform when a particular conformance tests fail.  With current implementation, TestEnvVars can fail due to either MUST or SHOULD not being conformant, better to decouple.

Reply:
> Regarding 1. would it be better I move the tests to a new folder runtime-conformance?

I think it's better using `conformance`. Once we have more than one runtime conformance test, we can move them all to `conformance/runtime` to keep things organized.

Reply:
Sounds good. I have moved the test into "conformance" folder along with breaking the tests that explicitly verify SHOULD and MUST.